core/store/src/genesis/initialization.rs

```diff
@@ -34,51 +34,47 @@ pub fn initialize_sharded_genesis_state(
     genesis_epoch_config: &EpochConfig,
     home_dir: Option<&Path>,
 ) {
-    // Ignore initialization if we already have genesis hash and state roots in store
```

Comment:
Does this comment no longer apply?

Reply:
So I removed storing the genesis json hash as this was the only place it was used. And for the longest of time it was kinda broken as we weren't really checking the hash at all. See the TODO below (#4447)

I changed the approach to now rely on the genesis state roots as a check for whether the genesis is initialized or not, which is quite a similar check as json hash.

We now also have other safeguards now like checking the hash of genesis block etc.